File: Database/java/main/com/cdedonder/amezzon/database/data/Transaction.java
package com.cdedonder.amezzon.database.data;

import com.cdedonder.amezzon.database.data.util.LocalDateTimeDeserializer;
import com.cdedonder.amezzon.database.data.util.LocalDateTimeSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

import java.io.IOException;
import java.time.LocalDateTime;

public class Transaction extends AbstractDTO {

    private int id;
    private TransactionType type;
    private Person client;
    private Product product;
    private double price;
    private int quantity;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime dateTime;

    public TransactionType getType() {
        return type;
    }

    public Person getClient() {
        return client;
    }

    public Product getProduct() {
        return product;
    }

    public double getPrice() {
        return price;
    }

    public int getQuantity() {
        return quantity;
    }

    public LocalDateTime getDateTime() {
        return dateTime;
    }

    public static Transaction fromJSON(String json) throws IOException {
        return objectMapper.readValue(json, Transaction.class);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setType(TransactionType type) {
        this.type = type;
    }

    public void setClient(Person client) {
        this.client = client;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public void setDateTime(LocalDateTime dateTime) {
        this.dateTime = dateTime;
    }

    @Override
    public String toJSON() throws IOException {
        return objectMapper.writeValueAsString(this);
    }
}

File: Client/domain/Transaction.py
from domain.GenericModel import GenericModel
from domain.Person import Person
from domain.TransactionType import TransactionType


class Transaction(GenericModel):
    def __init__(self, id=None, transaction_type=TransactionType(), person=Person(), price=None,
                 quantity=None, date_time=None):
        self.id = id
        self.transaction_type = transaction_type
        self.person = person
        self.price = price
        self.quantity = quantity
        self.date_time = date_time

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_transaction_type(self):
        return self.transaction_type

    def set_transaction_type(self, transaction_type):
        self.transaction_type = transaction_type

    def get_person(self):
        return self.person

    def set_person(self, person):
        self.person = person

    def get_quantity(self):
        return self.quantity

    def set_quantity(self, quantity):
        self.quantity = quantity

    def get_date_time(self):
        return self.date_time

    def set_date_time(self, date_time):
        self.date_time = date_time